src/common/com/intellij/plugins/haxe/model/HaxeClassModel.java

```diff
@@ -203,8 +203,9 @@ public SpecificHaxeClassReference getUnderlyingClassReference(HaxeGenericResolve
     for (HaxeIdentifier id : UsefulPsiTreeUtil.getChildren(haxeClass, HaxeIdentifier.class)) {
       if (id.getText().equals("to")) {
         PsiElement sibling = UsefulPsiTreeUtil.getNextSiblingSkipWhiteSpacesAndComments(id);
-        if (sibling instanceof HaxeType) {
-          types.add((HaxeType)sibling);
+        if (sibling instanceof HaxeTypeOrAnonymous) {
+          HaxeTypeOrAnonymous typeOrAnonymous = (HaxeTypeOrAnonymous) sibling;
```

Comment:
Isn't an anonymous type allowed in this position?  In such a case, the compiler would expect types that could unify with the anonymous type.  (I think.)

Reply:
Not sure i follow, it used to be HaxeType (only type), i have now changed it to typeOrAnonymous so that it can be both type  and anonymousType. The HaxeTypeOrAnonymous Psi element is also what the parser returns.

![image](https://user-images.githubusercontent.com/3193925/105631580-6d0e1980-5e4f-11eb-9bef-fe8a73e5fa63.png)

Reply:
What I was trying to point out is that you've opened up the parsing to the correct syntax (anonymous objects are allowed, Yay!), but the code doesn't handle it, because it only gets a type from a typeOrAnonymous (as `typeOrAnonymous.getType()` in the following line; it doesn't check if there's an anonymous type (e.g. `typeOrAnonymous.getAnonymousType()` that should be returned instead.

Reply:
It would also be nice to include an anonymous type in the test case below.